atom/browser/api/atom_api_app.cc

```diff
@@ -1154,6 +1154,7 @@ v8::Local<v8::Promise> App::GetFileIcon(const base::FilePath& path,
 
 std::vector<mate::Dictionary> App::GetAppMetrics(v8::Isolate* isolate) {
   std::vector<mate::Dictionary> result;
+  result.reserve(app_metrics_.size());
```

Comment:
I think this can be accomplished with the `vector( size_type count );` constructor, i.e.

```cpp
std::vector<mate::Dictionary> result(app_metrics_.size())
```

(in this and the other two locations)

Reply:
That is bad, because then you aren't just reserving enough space but also default-initializing those many values, the when instead reserve effects only the cap (not the size) so enough memory is reserved, but does not initialize them.

Reply:
Ah, yes, you're right.